src/Layout/redist/targets/GenerateTestingGlobalJson.targets

```diff
@@ -0,0 +1,32 @@
+<Project>
+  <PropertyGroup>
+    <_TestingGlobalJsonPath>$(ArtifactsTmpDir)global.json</_TestingGlobalJsonPath>
+  </PropertyGroup>
+
+  <!-- Since the dotnet binary respects sdk.paths in global.json now, and we use this in the repo root to make sure that we consistently use the repo-local
+       SDK for building/etc, we need to put something in place so that tests don't use the repo-local SDK and instead use the redist SDK. -->
+  <Target Name="CreateRedistGlobalJsonForTesting"
+    BeforeTargets="AfterBuild"
+    Outputs="$(_TestingGlobalJsonPath)">
```

Comment:
Hmm don't you need Inputs as well to make this fully incremental?

Reply:
Most of the actual incrementality is coming from the `WriteOnlyIfDifferent` on the Task